work/worker.go

```diff
@@ -533,11 +533,18 @@ func (self *worker) commitNewWork() {
 		// wait for a while and get a new timestamp
 		if tstart.Before(ideal) {
 			wait := ideal.Sub(tstart)
-			logger.Info("Mining too far in the future", "wait", common.PrettyDuration(wait))
+			logger.Debug("Mining too far in the future", "wait", common.PrettyDuration(wait))
 			time.Sleep(wait)
 
 			tstart = time.Now()    // refresh for metrics
 			tstamp = tstart.Unix() // refresh for block timestamp
+		} else {
+			logger.Info("Mining start for new block is later than expected",
+				"nextBlockNum", nextBlockNum,
+				"delay", tstamp-parent.Time().Int64()-1,
```

Comment:
1. I couldn't get why subtract 1 is added here. Could you explain it?
2. How about declaring `parentTime` in L531 and replacing L531,544,545?

Reply:
Since blocks are generated every second, each new block has a one-second delay. We subtract 1 from the total delay to focus on any extra delay beyond this one second to find the difference between the ideal and actual start of the next block.

Reply:
```suggestion
//  subtract 1 from the total delay to focus on any extra delay beyond this one second 
// to find the difference between the ideal and actual start of the next block.
				"delay", tstamp-parent.Time().Int64()-1,
```

Reply:
Thank you